pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1391,44 +1391,40 @@ public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
                                 PersistentTopic persistentTopic = isSystemTopic(topic)
                                         ? new SystemTopic(topic, ledger, BrokerService.this)
                                         : new PersistentTopic(topic, ledger, BrokerService.this);
-                                CompletableFuture<Void> preCreateSubForCompaction =
-                                        persistentTopic.preCreateSubscriptionForCompactionIfNeeded();
-                                CompletableFuture<Void> replicationFuture = persistentTopic
+                                persistentTopic
                                         .initialize()
-                                        .thenCompose(__ -> persistentTopic.checkReplication());
-
-                                CompletableFuture.allOf(preCreateSubForCompaction, replicationFuture)
-                                .thenCompose(v -> {
-                                    // Also check dedup status
-                                    return persistentTopic.checkDeduplicationStatus();
-                                }).thenRun(() -> {
-                                    log.info("Created topic {} - dedup is {}", topic,
+                                        .thenCompose(__ -> persistentTopic.preCreateSubscriptionForCompactionIfNeeded())
+                                        .thenCompose(__ -> persistentTopic.checkReplication())
+                                        .thenCompose(v -> {
+                                            // Also check dedup status
+                                            return persistentTopic.checkDeduplicationStatus();
```

Comment:
For convenient review, a little explanation here:
Move `preCreateSubscriptionForCompactionIfNeeded` after `initialize`.  because in `preCreateSubscriptionForCompactionIfNeeded` it will retrieve the compaction threshold, it could only get the value at the broker level. Only after `initialize`,  namespace and topic level values will load into the topic.  So it's the reason to do this.  Besides, I think  `preCreateSubscriptionForCompactionIfNeeded` should stay after `initialize` is more accurate .